src/main/java/org/mockito/internal/matchers/text/MatchersPrinter.java

```diff
@@ -31,7 +31,10 @@ public String getArgumentsBlock(List<ArgumentMatcher> matchers, PrintSettings pr
         int i = 0;
         for (final ArgumentMatcher matcher : matchers) {
             if (matcher instanceof ContainsExtraTypeInfo && printSettings.extraTypeInfoFor(i)) {
-                out.add(new FormattedText(((ContainsExtraTypeInfo) matcher).toStringWithType()));
+                out.add(new FormattedText(((ContainsExtraTypeInfo) matcher).toStringWithType(false)));
+            } else if(matcher instanceof ContainsExtraTypeInfo
+                      && printSettings.fullyQualifiedNameFor(((ContainsExtraTypeInfo) matcher).getWantedClass().getSimpleName())){
+                out.add(new FormattedText(((ContainsExtraTypeInfo) matcher).toStringWithType(true)));
```

Comment:
I think if we refactor it a bit, it is a bit more readable:

```java
if (matcher instanceof ContainsExtraTypeInfo) {
    ContainsExtraTypeInfo typeInfoMatcher = (ContainsExtraTypeInfo) matcher;
    Class<?> wantedClass = typeInfoMatcher.getWanted().getClass();
    String simpleNameOfArgument = wantedClass.getSimpleName();

    if (printSettings.extraTypeInfoFor(i)) {
        out.add(new FormattedText(typeInfoMatcher.toStringWithType(simpleNameOfArgument)));
    } else if (printSettings.fullyQualifiedNameFor(simpleNameOfArgument)) {
        out.add(new FormattedText(typeInfoMatcher.toStringWithType(wantedClass.getCanonicalName())));
    }
}
```

Reply:
Have refactored this, but also had to add the else to this block of instance of as well, because for the case when matcher is of type ContainsExtraTypeInfo but extra info and fully qualified name is also not needed. So if that else block is not included it would take them as empty string.